### PR TITLE
Add commands to add, update and delete users and loggers in MI instances

### PR DESCRIPTION
**Pull request by SanojPunchihewa**
## Purpose
Update MI CLI tool and merge with APICTL
wso2/micro-integrator#1999

## Goals
- Add a new add command under mi alias to add users and loggers to MI instances
- Add a new update command under mi alias to update log levels of loggers in MI instances
- Add a new delete command under mi alias to delete users from MI instances

## Approach
- Created a new package called add inside the mi package
- Add the necessary sub-commands(user and log-level) inside that get package
- Created a new package called delete inside the mi package and add sub-command user inside that get package
- Created a new package called update inside the mi package and add sub-command log-level inside that get package

## User stories
- To add a new user to a MI instance
`apictl mi add user [user-name]`
- To add a new logger to a MI instance
`apictl mi add log-level [logger-name] [class-name] [log-level]`
- To update the log level of a logger in MI instance
`apictl mi update log-level [logger-name] [log-level]`
- To delete a user from a MI instance
`apictl mi delete user [user-name]`

## Test environment
Ubuntu 20.04.1 LTS
Go version go1.15 linux/amd64

### CODE REVIEW

File: import-export-cli/cmd/mi/add/user.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+var addUserCmdEnvironment string
+
+const addUserCmdLiteral = "user [user-name]"
+const addUserCmdShortDesc = "Add new user to a Micro Integrator"
+
+const addUserCmdLongDesc = "Add a new user with the name specified by the command line argument [user-name] to a Micro Integrator in the environment specified by the flag --environment, -e"
+
+var addUserCmdExamples = "To add a new user\n" +
+	"  " + utils.ProjectName + " " + utils.MiCmdLiteral + " " + addCmdLiteral + " user user2 -e dev\n" +
```
Comment (wasuradananjith) [inline import-export-cli/cmd/mi/add/user.go:42]: Shall we have readable/meaningful username for the users? (Please check other occurrences as well)

File: import-export-cli/cmd/mi/delete/delete.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+	"github.com/spf13/cobra"
+	"github.com/wso2/product-apim-tooling/import-export-cli/utils"
+)
+
+const deleteCmdLiteral = "delete"
+const deleteCmdShortDesc = "Delete users from a Micro Integrator instance"
+
+const deleteCmdLongDesc = "Delete users from a Micro Integrator instance in the environment specified by the flag (--environment, -e)"
+
+const deleteCmdExamples = utils.ProjectName + " " + utils.MiCmdLiteral + " " + deleteCmdLiteral + " " + "user" + " user3 -e dev"
```
Comment (Chamindu36) [inline import-export-cli/cmd/mi/delete/delete.go:31]: Shall we change these user3 name into an example name?

File: import-export-cli/cmd/mi/delete/user.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+	credentials.HandleMissingCredentials(deleteUserCmdEnvironment)
+	executeDeleteUser(args[0])
+}
+
+func executeDeleteUser(userName string) {
+	resp, err := impl.DeleteMIUser(deleteUserCmdEnvironment, userName)
+	if err != nil {
+		fmt.Println(utils.LogPrefixError+"deleteing user [ "+userName+" ]", err)
+	} else {
+		fmt.Println("Deleteing user [ "+userName+" ] status:", resp)
```
Comment (Chamindu36) [inline import-export-cli/cmd/mi/delete/user.go:69]: Shall we change this as follows?
```suggestion
		fmt.Println("Deleting user [ "+userName+" ] status:", resp)
```

File: import-export-cli/cmd/mi/add/user.go (hunk 1 above)
Comment (SanojPunchihewa) [reply import-export-cli/cmd/mi/add/user.go]: Fixed with 3e012c9

File: import-export-cli/cmd/mi/delete/delete.go (hunk 2 above)
Comment (SanojPunchihewa) [reply import-export-cli/cmd/mi/delete/delete.go]: Fixed with 3e012c9

File: import-export-cli/cmd/mi/delete/user.go (hunk 3 above)
Comment (SanojPunchihewa) [reply import-export-cli/cmd/mi/delete/user.go]: Fixed with 3e012c9

**Review by wasuradananjith — APPROVED** [review]

**Review by 1akshitha — CHANGES_REQUESTED** [review]

File: import-export-cli/mi/impl/user.go [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+func deleteMIUser(url, env string) (string, error) {
+	resp, err := invokeDELETERequestWithRetry(url, env)
+	return handleResponse(resp, err, url, "status", "Error")
+}
+
+func resolveIsAdmin(isAdminConsoleInput string) string {
+	if len(strings.TrimSpace(isAdminConsoleInput)) == 0 {
+		return "false"
+	}
+	yesResponses := []string{"y", "yes"}
```
[the remark this answers is not in the document]
Comment (SanojPunchihewa) [reply import-export-cli/mi/impl/user.go:69]: We are using the [EqualFold](https://golang.org/pkg/strings/#EqualFold) method in go to compare the strings which is case-insensitive

File: import-export-cli/cmd/mi/add/logLevel.go [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+func handleAddLogLevelCmdArguments(args []string) {
+	printAddCmdVerboseLog("log-level")
+	credentials.HandleMissingCredentials(addLogLevelCmdEnvironment)
+	executeAddNewLogger(args[0], args[1], args[2])
+}
+
+func executeAddNewLogger(loggerName, logClass, logLevel string) {
+	resp, err := impl.AddMILogger(addLogLevelCmdEnvironment, loggerName, logClass, logLevel)
+	if err != nil {
+		fmt.Println(utils.LogPrefixError+"Adding new logger [ "+loggerName+" ] ", err)
```
[the remark this answers is not in the document]
Comment (SanojPunchihewa) [reply import-export-cli/cmd/mi/add/logLevel.go:67]: I used go formatting which was already in the project and it removes the space before and after the "+" automatically.

**Review by 1akshitha — APPROVED** [review]

**Pull request merged**